values-schema.yaml

```diff
@@ -2292,11 +2295,14 @@ properties:
                       client_secret:
                         type: string
                         x-secret: ''
+                      key_name:
+                        $ref: '#/definitions/wordCharacterPattern'
                     required:
                       - vault_name
                       - tenant_id
                       - client_id
                       - client_secret
+                      - key_name
```

Comment:
any reason to make this required? I think it should be optional

Reply:
because it throws an error if you don't include a key name and is required by the Vault schema
from the issue:
`Error parsing Seal configuration: key name is required`

Reply:
But I mean it is optional IF you don't want to opt into your new unseal setup. So it is ONLY required then. If you look into other areas where we have opional required fields (look for anyOf) you understand. We try to keep the api schema in core well validated for `otomi validate-values`.
@j-zimnowoda please also check these situations in reviews.

Reply:
Please review the code and note that all fields are required for this feature to work, and that this PR is only in the context of Azure KeyVault.

Reply:
I meant this comment that was created by me, that was set to resolved by you. I get that you thought I would not respond in time, but I still don't see why this breaking change was introduced for Azure, when the signing key should have been made optional. @j-zimnowoda why was this not caught?

Reply:
Again, the key name is required. Please read #845 as linked in the PR:
`Error parsing Seal configuration: key name is required`
No breaking change was introduced as the feature was broken to begin with.

Reply:
Also, this was all in consultation with Jeho while you were on holidays. Just because I set it to resolved doesn't mean I made that decision on my own.

Reply:
That is not true, as we were using already Azure signing for a long time. Also on GemU ONT and ACC. I use it all the time. It is simply not mandatory to determine a key, and Azure will generate one for you.